### ISSUE TITLE
budget change tracking

### ISSUE THREAD

**Issue by virithavanama** [google/lightweight_mmm#146]
Hi Team,

when we use a find_optimal_budgets, how to get the past budget for the n_time_periods vs the newly selected/given budget

**Comment by statm3n** [comment]
Hi @virithavanama ,

You could refer to the optimization section (starting from cell 26) in the [example notebook](https://github.com/google/lightweight_mmm/blob/main/examples/end_to_end_demo_with_multiple_geos.ipynb). Specifically you could find calculations of previous budget allocation and optimal budget allocation in cell 28 and cell 29. 

Note: find_optimal_budges performs optimization across the n_time_periods and under the same budget bound of your choice.

**Comment by virithavanama** [comment]
This gives the new budget split old vs new but what I want is for this period what is the previous mean budget for the model

**Comment by statm3n** [comment]
Hi, @virithavanama. Thanks for clarifying the question. would cell 27 in [example notebook](https://github.com/google/lightweight_mmm/blob/main/examples/end_to_end_demo_with_multiple_geos.ipynb) help?

If you are interested in how the package calculates previous mean media(budget). You could dig into optimize_media.py.[line 131](https://github.com/google/lightweight_mmm/blob/c7627923175b69360f3906951a9e5e62f69dd1d4/lightweight_mmm/optimize_media.py#L131), [line 172](https://github.com/google/lightweight_mmm/blob/c7627923175b69360f3906951a9e5e62f69dd1d4/lightweight_mmm/optimize_media.py#L172), and [line 179 ](https://github.com/google/lightweight_mmm/blob/c7627923175b69360f3906951a9e5e62f69dd1d4/lightweight_mmm/optimize_media.py#L179). 

The package has a few assumptions for future budget optimization. Generally speaking, previous budget = previous impressions * previous prices.

**Comment by statm3n** [comment]
I would detour you to the [lightweightmmm class](https://github.com/google/lightweight_mmm/blob/c7627923175b69360f3906951a9e5e62f69dd1d4/lightweight_mmm/lightweight_mmm.py#L121) to check saved attributes in the model object.

Unfortunately, saved model object does not have a method to output historical budget. You'll have to code a bit. You'll also need to make a few assumptions to carry out the calculations. 

We will keep an eye on similar asks regarding historical budget. If this is common, we should consider add it to the package.

**Comment by virithavanama** [comment]
Yes please keep this in the queue because this will help in comparing the budget change similar to the previous and optimized channel-wise split and conversions change
